File: Umbraco.Web/Models/ContentEditing/SaleOrderDisplay.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Umbraco.Web.Models.ContentEditing
{
    public class SaleOrderDisplay
    {
        public SaleOrderDisplay()
        {
            State = "draft";
            DateOrder = DateTime.Now;
            Name = "/";
            AmountTotal = 0;
        }
        public Guid Id { get; set; }

        /// <summary>
        /// Ngày điều trị
        /// </summary>
        public DateTime DateOrder { get; set; }

        /// <summary>
        /// Khách hàng
        /// </summary>
        public Guid PartnerId { get; set; }
        public PartnerSimple Partner { get; set; }

        public decimal? AmountTax { get; set; }

        public decimal? AmountUntaxed { get; set; }

        public decimal? AmountTotal { get; set; }
        public decimal? PaidTotal
        {
            get
            {
                return (this.AmountTotal ?? 0) - (this.Residual ?? 0);
            }
            set { }
        }

        public string Note { get; set; }

        /// <summary>
        /// Trạng thái
        /// </summary>
        public string State { get; set; }

        public string Name { get; set; }

        public IEnumerable<SaleOrderLineDisplay> OrderLines { get; set; } = new List<SaleOrderLineDisplay>();



        public Guid CompanyId { get; set; }

        /// <summary>
        /// Nhân viên, bác sĩ điều trị
        /// </summary>
        public string UserId { get; set; }
        public ApplicationUserSimple User { get; set; }

        public decimal? Residual { get; set; }
        public decimal? Paid { get; set; }

        public Guid? PricelistId { get; set; }
        public ProductPricelistBasic Pricelist { get; set; }

        public Guid? JournalId { get; set; }
        public AccountJournalSimple Journal { get; set; }

        public bool? IsQuotation { get; set; }
        public bool? IsFast { get; set; }

        public SaleOrderBasic Quote { get; set; }

        public SaleOrderBasic Order { get; set; }

        public string InvoiceStatus { get; set; }

        public int InvoiceCount { get; set; }
    }

    public class SaleOrderDisplayVm
    {
        public SaleOrderDisplayVm()
        {
            State = "draft";
            DateOrder = DateTime.Now;
            Name = "/";
            AmountTotal = 0;
        }
        public Guid Id { get; set; }

        /// <summary>
        /// Ngày điều trị
        /// </summary>
        public DateTime DateOrder { get; set; }

        /// <summary>
        /// Khách hàng
        /// </summary>
        public Guid PartnerId { get; set; }
        public PartnerDisplay Partner { get; set; }

        public decimal? AmountTax { get; set; }

        public decimal? AmountUntaxed { get; set; }

        public decimal? AmountTotal { get; set; }
        public decimal? PaidTotal
        {
            get
            {
                return (this.AmountTotal ?? 0) - (this.Residual ?? 0);
            }
            set { }
        }

        public string Note { get; set; }

        /// <summary>
        /// Trạng thái
        /// </summary>
        public string State { get; set; }

        public string Name { get; set; }

        public IEnumerable<SaleOrderLineDisplay> OrderLines { get; set; } = new List<SaleOrderLineDisplay>();

        public IEnumerable<DotKhamDisplayVm> DotKhams { get; set; } = new List<DotKhamDisplayVm>();

        public IEnumerable<SaleOrderPromotionBasic> Promotions { get; set; } = new List<SaleOrderPromotionBasic>();

        public Guid CompanyId { get; set; }

        /// <summary>
        /// Nhân viên, bác sĩ điều trị
        /// </summary>
        public string UserId { get; set; }
        public ApplicationUserSimple User { get; set; }

        public decimal? Residual { get; set; }
        public decimal? Paid { get; set; }

        public Guid? PricelistId { get; set; }
        public ProductPricelistBasic Pricelist { get; set; }

        public Guid? JournalId { get; set; }
        public AccountJournalSimple Journal { get; set; }

        public bool? IsQuotation { get; set; }
        public bool? IsFast { get; set; }

        public SaleOrderBasic Quote { get; set; }

        public SaleOrderBasic Order { get; set; }

        public string InvoiceStatus { get; set; }

        public int InvoiceCount { get; set; }
    }
}

File: TMTDentalAPI/ClientApp/src/app/sale-orders/sale-order-display.ts
import { AccountJournalSimple } from 'src/app/account-journals/account-journal.service';
import { EmployeeSimple } from '../employees/employee';
import { PartnerSimple } from '../partners/partner-simple';
import { ProductPriceListBasic } from '../price-list/price-list';
import { QuotationSimple } from '../quotations/quotation.service';
import { UserSimple } from '../users/user-simple';
import { SaleOrderBasic } from './sale-order-basic';
import { SaleOrderLineDisplay } from './sale-order-line-display';

export class SaleOrderDisplay {
    id: string;
    partner: PartnerSimple;
    partnerId: string;
    dateOrder: string;
    user: UserSimple;
    employee: EmployeeSimple;
    employeeId: string;
    companyId: string;
    userId: string;
    state: string;
    residual: number;
    orderLines: SaleOrderLineDisplay[];
    name: string;
    paid: number;
    note: string;
    pricelist: ProductPriceListBasic;
    journalId: string;
    journal: AccountJournalSimple;
    invoiceStatus: string;
    quote: SaleOrderBasic;
    order: SaleOrderBasic;
    invoiceCount: number;
    paidTotal: number;
    quotationId: string;
    quotation: QuotationSimple;
    promotions: any[];
    totalPaid: number;
}

export class SaleOrderDisplayVm {
    id: string;
    partner: any;
    partnerId: string;
    dateOrder: string;
    user: UserSimple;
    employee: EmployeeSimple;
    employeeId: string;
    companyId: string;
    userId: string;
    state: string;
    residual: number;
    orderLines: any[];
    dotKhams: any[];
    name: string;
    paid: number;
    pricelist: ProductPriceListBasic;
    journalId: string;
    journal: AccountJournalSimple;
    invoiceStatus: string;
    quote: SaleOrderBasic;
    order: SaleOrderBasic;
    invoiceCount: number;
    paidTotal: number;
}